docs/machine-learning/how-to-guides/prepare-data-ml-net.md

```diff
@@ -0,0 +1,290 @@
+
+## Filter data
+
+Sometimes, not all data in a dataset is relevant for analysis. An approach to remove irrelevant data is filtering.
+
+Using the following input data which is loaded into an [`IDataView`](xref:Microsoft.ML.IDataView):
+
+```csharp
+HomeData[] homeDataList = new HomeData[]
+{
+    new HomeData
+    {
+        NumberOfBedrooms=1f,
+        Price=100000f
+    },
+    new HomeData
+    {
+        NumberOfBedrooms=2f,
+        Price=300000f
+    },
+    new HomeData
+    {
+        NumberOfBedrooms=6f,
+        Price=600000f
+    }
+};
+```
+
+To filter data based on the value of a column, use the [`FilterRowsByColumn`](xref:Microsoft.ML.DataOperationsCatalog.FilterRowsByColumn*) method.
+
+```csharp
+// Apply filter
+IDataView filteredData = mlContext.Data.FilterRowsByColumn(data, "Price", lowerBound: 200000, upperBound: 1000000);
+```
+
+The sample above takes rows in the dataset with a price between 200000 and 1000000. The result of applying this filter would return only the last two rows in the data and exclude the first row because its price is 100000 and not between the specified range.
+
+## Use normalizers
+
+[Normalization](https://en.wikipedia.org/wiki/Feature_scaling) is a data pre-processing technique used to standardize features that are not on the same scale. For example, the ranges for values like age and income vary significantly with age generally being in the range of 0-100 and income generally being in the range of zero to thousands. Visit the [transforms page](../resources/transforms.md) for a more detailed list and description of normalization transforms. 
+
+### Min-Max normalization
+
+Using the following input data which is loaded into an [`IDataView`](xref:Microsoft.ML.IDataView):
+
+```csharp
+HomeData[] homeDataList = new HomeData[]
+{
+    new HomeData
+    {
+        NumberOfBedrooms = 2f,
+        Price = 200000f
+    },
+    new HomeData
+    {
+        NumberOfBedrooms = 1f,
+        Price = 100000f
+    }
+};
+```
+
+Normalize the data using min-max normalization using the [`NormalizeMinMax`](xref:Microsoft.ML.NormalizationCatalog.NormalizeMinMax*) method.
+
+```csharp
+// Define min-max estimator
+var minMaxEstimator = mlContext.Transforms.NormalizeMinMax("Price");
+
+// Build ITransformer that performs operation
+ITransformer minMaxTransformer = minMaxEstimator.Fit(data);
+
+// Transform data
+IDataView transformedData = minMaxTransformer.Transform(data);
+```
+
+The original price values `[200000,100000]` are converted to `[ 1, 0.5 ]` using the `MinMax` normalization formula.
+
+### Binning
+
+[Binning](https://en.wikipedia.org/wiki/Data_binning) converts continuous values into a discrete representation. For example, suppose one of your features is age. Instead of using the actual age value, with binning you can create ranges for that value. 0-18 could be one bin, another could be 19-35 and so on. 
```

Comment:
>create ranges for that value. 0-18 could be one bin, another could be 19-35 and so on. [](start = 223, length = 86)

This might give them the illusion that they can control the bin upperbounds, which they can't... they can just control the number of bins..